src/fitnesse/testsystems/slim/SlimTestSystem.java

```diff
@@ -118,83 +137,114 @@ protected SlimTestContextImpl createTestContext(TestPage testPage) {
 
   protected void processTable(SlimTable table, boolean isSuiteTearDownPage) throws TestExecutionException {
     List<SlimAssertion> assertions = table.getAssertions();
-    final Map<String, Object> instructionResults;
+     Map<String, Object> instructionResults;
     if (stopTestCalled && !table.isTearDown()) {
       instructionResults = Collections.emptyMap();
+      evaluateTables(assertions, instructionResults);
     } else if (ignoreAllTestsCalled && !table.isTearDown()){
       instructionResults = Collections.emptyMap();
+      evaluateTables(assertions, instructionResults);
     } else {
       boolean tearDownOfAlreadyStartedTest = stopTestCalled && table.isTearDown();
       if(ignoreAllTestsCalled && table.isTearDown()){
         ignoreAllTestsCalled = false;
       }
       if (stopSuiteCalled && !isSuiteTearDownPage && !tearDownOfAlreadyStartedTest) {
         instructionResults = Collections.emptyMap();
+        evaluateTables(assertions, instructionResults);
       } else {
-        instructionResults = slimClient.invokeAndGetResponse(SlimAssertion.getInstructions(assertions));
+        if (isModeSingle()){
```

Comment:
This behaviour also requires a **unit** test to see that each assertion triggers `testAssertionWillBeExecuted()`, `slimClient.invokeAndGetResponse()` and `evaluateAssertion()` with the right parameters.